src/task.c

```diff
@@ -313,6 +313,10 @@ void JL_NORETURN jl_finish_task(jl_task_t *ct)
 {
     JL_PROBE_RT_FINISH_TASK(ct);
     JL_SIGATOMIC_BEGIN();
+    if (ct->metrics_enabled) {
+        assert(ct->first_enqueued_at != 0);
+        ct->finished_at = jl_hrtime();
```

Comment:
```suggestion
        assert(jl_atomic_load_relaxed(ct->first_enqueued_at) != 0);
        jl_atomic_store_relaxed(&ct->finished_at, jl_hrtime());
```

Reply:
Think you need an `&` on the relaxed load (in the `assert`).